[SandboxIR] Add ExtractValueInst.

slackito opened the pull request.

slackito committed:
[SandboxIR] Add ExtractValueInst.

vporpo commented on llvm/include/llvm/SandboxIR/SandboxIR.h:
```diff
@@ -1621,6 +1626,61 @@ class UnaryInstruction
+    return From->getSubclassID() == ClassID::ExtractValue;
+  }
+
+  static Type *getIndexedType(Type *Agg, ArrayRef<unsigned> Idxs) {
```
nit: It is not straight forward what this function does, so I would copy LLVM IR's comments.

slackito replied:
Done.

vporpo commented on llvm/unittests/SandboxIR/SandboxIRTest.cpp:
```diff
@@ -1261,6 +1261,102 @@ define void @foo(<2 x i8> %v1, <2 x i8> %v2) {
+  EXPECT_EQ(sandboxir::ExtractValueInst::getIndexedType(
+                AggType, ArrayRef<unsigned>({1, 0})),
+            llvm::ExtractValueInst::getIndexedType(AggType,
+                                                   ArrayRef<unsigned>({1, 0})));
```
Could you have two checks for `getIndexedType()`, one that returns non-null and one that returns null. The reason is that once we migrate to `sandboxir::Type` the null case would need special treatment.

slackito replied:
Added a simple test for the nullptr case. Please take another look.

slackito committed:
Address review feedback.

- Copy comment from llvm::ExtractValueInst::getIndexedType
- Add test case for getIndexedType returning nullptr

vporpo reviewed (approved):
LG thanks!
